lessons/flow.md

```diff
@@ -112,21 +149,207 @@ No errors!
+capitalized. Therefore, the type `Number` is not the same as the type `number`. The full list of primitive types is:
+
+* [`Booleans`](https://flow.org/en/docs/types/primitives/#toc-booleans)
+* [`Number`](https://flow.org/en/docs/types/primitives/#toc-numbers)
+* [`String`](https://flow.org/en/docs/types/primitives/#toc-strings)
+
+#### 3.1.2 [`null` vs. `void`](https://flow.org/en/docs/types/primitives/#toc-null-and-void)
+
+The type of `null` is `null` but the type of `undefined` is `void`. Therefore, a function which input is a number and 
+returns `undefined`, its type will be `(number) => void`.
+
+#### 3.1.3 [Maybe types](https://flow.org/en/docs/types/primitives/#toc-maybe-types)
+
+In order to define a maybe type, you need put a `?` before the type. In this way, the type defined, `void` and `null` 
+will be accepted.
+
+For example: 
+```flow js
+function foo (x: ?number) {
+  //....
+}
+
+```
+It will accept any number, `null` or `undefined` as its input. 
+
+#### 3.1.4 Optional [object parameters](https://flow.org/en/docs/types/primitives/#toc-optional-object-properties) and [function parameters](https://flow.org/en/docs/types/primitives/#toc-optional-function-parameters)
+
+It is possible to declare a object property of a function argument by placing the `?` right after the variable name.
+
+```flow js
+function foo(value?: string) {
+  // ...
+}
+``` 
+
+`value` can be any sting and `undefined` but **not `null`**.
+
+### 3.2 [Mixed types](https://flow.org/en/docs/types/mixed/)
+
+In Flow we use Mixed types to indicates that a certain variable or argument can accept any kind of type. If you would like
+to interact with a variable which is `mixed` you need to ensure first that certain variable has a defined type.
+
+
+```flow js
+function stringify(value: mixed) {
+  if (typeof value === 'string') {
+    return "" + value; // Works!
+  } else {
+    return "";
+  }
+}
+```
+
+> Warning: Do not confused Mixed types with [`Any`](https://flow.org/en/docs/types/any/). **You should not use `Any`.** 
+
+### 3.3 [Function types](https://flow.org/en/docs/types/functions/)
+
+#### 3.3.1 [Return type of a function](https://flow.org/en/docs/types/functions/#toc-function-returns)
+
+We can add the return type of a function, by typing after the list of arguments.
+```flow js
+function foo(bar: string): number {
+  return 5;
+}
+```
+
+```flow js
+const method = (bar: string):number => {
+  return 5;
+}
+```
+
+#### 3.3.2 [Rest parameters](https://flow.org/en/docs/types/functions/#toc-rest-parameters)
+
+It is possible to type the rest parameters sintax as an `Array`.
+
+```flow js
+function method(...args: Array<number>) {
+  // ...
+}
+```
+
+####  3.3.3 [Arbitraty functions](https://flow.org/en/docs/types/functions/#toc-function-type)
+In the case you would like to express an arbitrary function, you should write `() => mixed`.
+
+> Warning: Do not use the `Function` type. Use `()=> mixed` instead. 
```

Comment:
Is this because `Function` is implicitly an `() => any`? Could you mention why not to use it?